vowpalwabbit/arguments.cc

```diff
@@ -0,0 +1,15 @@
+#include "arguments.h"
+
+using namespace VW;
+
+bool VW::operator==(const base_argument& lhs, const base_argument& rhs) {
+  return lhs.m_name == rhs.m_name
```

Comment:
It seems semantically confusing to have a top-level operator==() which just deals arguments.  Maybe this should be scoped to an arguments type?

Reply:
I think it is worth being able to check equality of two base_arguments, even though the default value can't be checked at this point

Reply:
I'm not questioning the usefulness of the function, but rather it's scope.

Reply:
There exists an operator for both the base_argument type as well as typed_argument<T>, so that the best operator can be used depending on how much type information you have at the time. I may be misunderstanding the question though

Reply:
Why should both operators be at the VW:: scope?

Reply:
They can be in the config name space